sdcm/utils/data_validator.py

```diff
@@ -493,90 +684,77 @@ def validate_range_expected_to_change(self, session, during_nemesis=False):
-                            f"See error above in the sct.log"
-                ).publish()
-                return
+            data_for_validation = self.fetch_data_for_validation_after_update(during_nemesis=during_nemesis,
+                                                                              views_set=views_set,
+                                                                              session=session)
+            if data_for_validation is None:
+                continue
 
             # Issue https://github.com/scylladb/scylla/issues/6181
             # Not fail the test if unexpected additional rows where found in actual result table
-            if len(before_update_rows) + len(after_update_rows) > len(expected_rows):
+            if len(data_for_validation.actual_data) > len(data_for_validation.expected_data):
                 DataValidatorEvent.UpdatedRowsValidator(
                     severity=Severity.WARNING,
                     message=f"View {views_set[0]}. "
-                            f"Actual dataset length {len(before_update_rows) + len(after_update_rows)} "
-                            f"more then expected dataset length: {len(expected_rows)}. "
+                            f"Actual dataset length {len(data_for_validation.actual_data)} "
+                            f"more then expected dataset length: {len(data_for_validation.expected_data)}. "
                             f"Issue #6181"
                 ).publish()
+                continue
+
+            if during_nemesis:
+                LOGGER.debug('Validation updated rows.  View %s. Actual dataset length %s, '
+                             'Expected dataset length: %s.',
+                             data_for_validation.views[0], len(data_for_validation.actual_data),
+                             len(data_for_validation.expected_data))
+                continue
+
+            if (len(data_for_validation.actual_data) != len(data_for_validation.expected_data)) \
+                    or data_for_validation.actual_data != data_for_validation.expected_data:
+                LOGGER.debug("%s. Rows amount:\n  before update: %s\n  after update: %s\n  expected: %s\n "
+                             "actual: %s",
+                             data_for_validation.views[0], len(data_for_validation.before_update_rows),
+                             len(data_for_validation.after_update_rows), len(data_for_validation.expected_data),
+                             len(data_for_validation.actual_data))
+
+                logdir = self.save_data_for_debugging(data_for_validation)
+
+                self.analyze_updated_data_and_save_in_file(data_for_validation=data_for_validation,
+                                                           session=session,
+                                                           logdir=logdir)
+
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.CRITICAL,
```

Comment:
why there's two critical event being raised ?

one in `analyze_updated_data_and_save_in_file` and one here ?

Reply:
but still we have two critical, that would be called one after anther.

as for getting system tables, if that would help debugging, we might consider extracting it too, on failures.

Reply:
I know @soyacz is tasked with something similar...

Reply:
Yes, I'm working on feature to collect selected keyspace/tables sstables on specific events and link them in report/argus. Work is in progress.

Reply:
Second CRITICAL event is redundant. Remove it.

@soyacz do you have opened PR for that or not yet? 
I can add `TODO` and when the feature (collect selected keyspace/tables sstables ) will be ready, we will decrease event severity to ERROR